### PR TITLE
feat(users): migrate userSignUp react query

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

The last one 🎉 

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-977

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview 28102cd2-684e-4ecb-a4f7-2f37f653b027 | [my-helm-env](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/28102cd2-684e-4ecb-a4f7-2f37f653b027/services/general) |
 /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Programs must be written for people to read, and only incidentally for machines to execute

File: libs/domains/cloud-providers/data-access/src/lib/domains-cloud-providers-data-access.ts [hunk 1]
```diff
@@ -8,12 +8,13 @@ import {
   type KubernetesEnum,
   type ScalewayCredentialsRequest,
 } from 'qovery-typescript-axios'
-import { type DistributiveOmit } from 'react-redux'
 import { match } from 'ts-pattern'
 
 const cloudProviderApi = new CloudProviderApi()
 const cloudProviderCredentialsApi = new CloudProviderCredentialsApi()
 
+type DistributiveOmit<T, K extends keyof T> = T extends unknown ? Omit<T, K> : never
```
Comment (ctjhoa) [inline libs/domains/cloud-providers/data-access/src/lib/domains-cloud-providers-data-access.ts:16]: use 
```
import { type DistributiveOmit } from '@qovery/shared/util-types'
```
instead.

File: libs/pages/onboarding/src/lib/feature/onboarding-more/onboarding-more.tsx [hunk 2]
```diff
@@ -51,43 +47,51 @@ const dataQuestions: Value[] = [
...
 
-      if (user.dx_auth) {
-        navigate(`${ONBOARDING_URL}${ONBOARDING_PROJECT_URL}`)
-      } else if (!user.dx_auth) {
-        // redirect to Thanks page if user not authorized by the dx team
-        // dx_auth must be updated in the bdd
-        navigate(`${ONBOARDING_URL}${ONBOARDING_THANKS_URL}`)
+        if (newUserSignUp?.dx_auth) {
+          navigate(`${ONBOARDING_URL}${ONBOARDING_PROJECT_URL}`)
+        } else if (!userSignUp?.dx_auth) {
```
Comment (ctjhoa) [inline libs/pages/onboarding/src/lib/feature/onboarding-more/onboarding-more.tsx:88]:
```suggestion
        } else {
```
?

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/onboarding/src/lib/feature/onboarding-more/onboarding-more.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/pages/onboarding/src/lib/feature/onboarding-more/onboarding-more.tsx]: https://github.com/Qovery/console/pull/1124/commits/548c6c007285939fd53eb89c1fd78345b0bdbff7

File: libs/domains/cloud-providers/data-access/src/lib/domains-cloud-providers-data-access.ts (hunk 1 above)
Comment (RemiBonnet) [reply libs/domains/cloud-providers/data-access/src/lib/domains-cloud-providers-data-access.ts]: https://github.com/Qovery/console/pull/1124/commits/548c6c007285939fd53eb89c1fd78345b0bdbff7

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**